packages/gatsby/src/schema/__tests__/rebuild-schema.js

```diff
@@ -77,24 +79,36 @@ const deleteNodeAndRebuild = async node => {
 const createExternalSchema = () => {
   const query = new GraphQLObjectType({
     name: `Query`,
-    fields: {
-      external: {
-        type: new GraphQLObjectType({
-          name: `ExternalType`,
-          fields: {
-            externalFoo: {
-              type: GraphQLString,
-              resolve: parentValue =>
-                `${parentValue}.ExternalType.externalFoo.defaultResolver`,
+    fields: () => {
```

Comment:
Is this to simulate the what the actual workaround is doing (using thunk/lazy fields)? Shouldn't we actually have "un-thunked" fields here (as before the PR) to actually test that workaround works?

Reply:
It doesn't matter actually because of `assertValidSchema(schema)` below that will resolve all thunks effectively anyway (it traverses the full schema).

And the funny part is that it was a missing piece when I was trying to reproduce the issue in tests. Tests were just refusing to fail. And then I had to do another pass of debugging just to figure out that in the real-world situation we have `extracting queries` step between schema builds.

And this step calls this `assertValidSchema` under the hood and that's how the stage gets set for the actual bug. It is not reproducible without it.